website/project/views/comment.py

```diff
@@ -15,104 +15,204 @@
 from website.notifications.emails import notify
 from website.filters import gravatar
 from website.models import Guid, Comment
+from website.addons.base import GuidFile
 from website.project.decorators import must_be_contributor_or_public
 from datetime import datetime
 from website.project.model import has_anonymous_link
+from website.project.views.node import _view_project, n_unread_comments
 
 
-def resolve_target(node, guid):
+@must_be_contributor_or_public
+def view_comments_project(auth, **kwargs):
+    """
+    Returns information needed to get comments for the total discussion page
+    """
+    OVERVIEW = "overview"
+    FILES = "files"
+    WIKI = "wiki"
+
+    node = kwargs['node'] or kwargs['project']
+    page = request.args.get('page', None)
+
+    if not page:
+        ret = _view_comments_total()
+    elif page == OVERVIEW:
+        ret = _view_comments_overview(auth, node)
+    elif page == FILES:
+        ret = _view_comments_files(auth, node)
+    elif page == WIKI:  # Wiki
+        ret = _view_comments_wiki(auth, node)
+    else:
```

Comment:
@sloria See line 37 <code>if not page:</code>
This line is to prevent something like this happening: <code>\<pid\>/discussions?page=sloria</code>